src/main/linking_flags.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+set -ue
+
+# This script is called by dune to generate the linking flags for static builds
+# (on the limited set of supported platforms). It only returns an empty set of
+# flags for the default dynamic linking mode.
+
+LC_ALL=C
+
+echo ";; generated by $0"
+
+case "$1" in
+    dynamic) echo "()"; exit 0;;
+    static) ;;
+    *) echo "Invalid linking mode '$1'. Usage: $0 dynamic|static [extra-libs]" >&2; exit 2
+esac
+
+shift
+EXTRA_LIBS="$*"
+
+## Static linking configuration ##
+
+# The linked C libraries list may need updating on changes to the dependencies.
+#
+# To get the correct list for manual linking, the simplest way is to set the
+# flags to `-verbose`, while on the normal `autolink` mode, then extract them
+# from the gcc command-line.
```

Comment:
Thanks for documenting this.

To ease further updates, do you believe it's feasible to partially automate this? (e.g., adding in the PR a special dune rule and/or make target to print the implied gcc command lines, if an ad-hoc environment variable is set to true)

Reply:
Re-running the (standard) build with `OCAMLPARAM=_,verbose=1` should be enough, find the related `gcc` command towards the end and extract the `-lXXX` arguments.